### DISCUSSION TITLE
Wireguard mode for LAN traffic

### DISCUSSION THREAD

**Discussion by atxsinn3r** [mitmproxy/mitmproxy#5721] [Q&A]
Hi, this may be a silly novice question. So I have two VMs, one of them is a server, and the other is a client. I notice that when I'm on wireguard mode, the traffic between these two do not go through the VPN interface. For example, my set up a nectar listener on VM A, and VM B connects to it, and mitmproxy does not pick up anything. Same for UDP. My question is how do I use mitmproxy's wireguard mode to see this traffic and intercept them? Thanks.

**Comment by mhils** [comment]
I think this depends on the network configuration of your connection-initiating VM. You want to configure your routing so that everything goes through the WireGuard interface, see https://www.wireguard.com/netns/ for example. It's not really a mitmproxy issue if that makes sense.

**Comment by atxsinn3r** [comment]
Thank you for the link. Yep you're right, it's not really a mitmproxy issue.
